test/unit/TransactionRecord.js

```diff
@@ -11,4 +11,37 @@ describe("TransactionRecord", function () {
+
+        const nftTransferMap = new TokenNFTTransferMap()
+        nftTransferMap.__set(TokenId.fromString("0.0.123"), {
+            sender: AccountId.fromString("0.0.456"),
+            recipient: AccountId.fromString("0.0.789"),
+            serial: 123,
+            isApproved: true
+        });
+
+        const newRecord = new TransactionRecord({
+            ...record,
+            transactionMemo: "test",
+            tokenTransfers: tokenTransferMap,
+            nftTransfers: nftTransferMap,
+            parentConsensusTimestamp: record.consensusTimestamp,
+            aliasKey: PublicKey.fromString("302a300506032b6570032100d7366c45e4d2f1a6c1d9af054f5ef8edc0b8d3875ba5d08a7f2e81ee8876e9e8"),
+            ethereumHash: Uint8Array.from([1, 2, 3, 4]),
+            paidStakingRewards: record.transfers,
+            prngBytes: Uint8Array.from([1, 2, 3, 4]),
+            prngNumber: 123,
+            evmAddress: EvmAddress.fromString("0xdeadbeef"),
+        });
+
+        const expectedJSON = JSON.parse(`{"receipt":{"status":"SUCCESS","accountId":"0.0.1246","filedId":null,"contractId":null,"topicId":null,"tokenId":null,"scheduleId":null,"exchangeRate":{"hbars":1,"cents":12,"expirationTime":"1963-11-25T17:31:44.000Z","exchangeRateInCents":12},"topicSequenceNumber":"0","topicRunningHash":{},"totalSupply":"0","scheduledTransactionId":null,"serials":[],"duplicates":[],"children":[]},"transactionHash":"cac44f2db045ba441f3fbc295217f2eb0f956293d28b3401578f6160e66f4e47ea87952d91c4b1cb5bda6447823b979a","consensusTimestamp":"2022-06-18T02:54:43.839Z","transactionId":"0.0.1157@1655520872.507983896","transactionMemo":"test","transactionFee":"41694270","transfers":[{"accountId":"0.0.5","amount":"1071080","isApproved":false},{"accountId":"0.0.98","amount":"32498552","isApproved":false},{"accountId":"0.0.800","amount":"4062319","isApproved":false},{"accountId":"0.0.801","amount":"4062319","isApproved":false},{"accountId":"0.0.1157","amount":"-1041694270","isApproved":false},{"accountId":"0.0.1246","amount":"1000000000","isApproved":false}],"tokenTransfers":{"0.0.123":{"0.0.456":789}},"tokenTransfersList":[],"scheduleRef":null,"assessedCustomFees":[],"nftTransfers":{"0.0.123":[{"sender":"0.0.456","recipient":"0.0.789","serial":123,"isApproved":true}]},"automaticTokenAssociations":[],"parentConsensusTimestamp":"2022-06-18T02:54:43.839Z","aliasKey":"302a300506032b6570032100d7366c45e4d2f1a6c1d9af054f5ef8edc0b8d3875ba5d08a7f2e81ee8876e9e8","duplicates":[],"children":[],"ethereumHash":"01020304","paidStakingRewards":[{"accountId":"0.0.5","amount":"1071080","isApproved":false},{"accountId":"0.0.98","amount":"32498552","isApproved":false},{"accountId":"0.0.800","amount":"4062319","isApproved":false},{"accountId":"0.0.801","amount":"4062319","isApproved":false},{"accountId":"0.0.1157","amount":"-1041694270","isApproved":false},{"accountId":"0.0.1246","amount":"1000000000","isApproved":false}],"prngBytes":"01020304","prngNumber":123,"evmAddress":"deadbeef"}`);
```

Comment:
Ideally those could be populated: `tokenTransfersList`, `tokenTransfersList`, `assessedCustomFees`, `automaticTokenAssociations`.